File: product/domain/category.go
package domain

type Category struct {
	ID       int64
	TenantID int64
	ParentID int64
	Name     string
	Level    int32
	Sort     int32
	Icon     string
	Status   CategoryStatus
	Children []Category
}

type CategoryStatus uint8

const (
	CategoryStatusActive CategoryStatus = 1
	CategoryStatusHidden CategoryStatus = 2
)

File: admin-frontend/src/api/category.ts
import { request } from './client'
import type { Category } from '@/types/product'

export async function listCategories() {
  return request<Category[]>({
    method: 'GET',
    url: '/categories',
  })
}

export interface CategoryData {
  parentId?: number
  name: string
  level?: number
  sort?: number
  icon?: string
  status?: number
}

export async function createCategory(data: CategoryData) {
  return request<{ id: number }>({
    method: 'POST',
    url: '/categories',
    data: { parent_id: data.parentId, name: data.name, level: data.level, sort: data.sort, icon: data.icon, status: data.status },
  })
}

export async function updateCategory(id: number, data: CategoryData) {
  return request<null>({
    method: 'PUT',
    url: `/categories/${id}`,
    data: { parent_id: data.parentId, name: data.name, level: data.level, sort: data.sort, icon: data.icon, status: data.status },
  })
}
